lib/Doctrine/DBAL/Query/QueryBuilder.php

```diff
@@ -1180,12 +1210,22 @@ private function isLimitQuery()
      * Converts this instance into an INSERT string in SQL.
      *
      * @return string
+     * @throws QueryException
      */
     private function getSQLForInsert()
     {
-        return 'INSERT INTO ' . $this->sqlParts['from']['table'] .
-        ' (' . implode(', ', array_keys($this->sqlParts['values'])) . ')' .
-        ' VALUES(' . implode(', ', $this->sqlParts['values']) . ')';
+        $statement = 'INSERT INTO ' . $this->sqlParts['into']['table'];
+        if( !empty($this->sqlParts['values']) && count( array_keys($this->sqlParts['values']) ) > 0 ) {
```

Comment:
Isn't it the same as `if (count($this->sqlParts['values']) > 0)`?

Reply:
that's right

Reply:
the `! empty()` is sufficient